src/details/ArborX_AccessTraits.hpp

```diff
@@ -103,6 +103,26 @@ struct AccessTraitsHelper<AccessTraits<X, Tag>>
   using tag = Kokkos::detected_t<PredicateTagArchetypeAlias, type>;
 };
 
+template <typename X, typename Tag>
+class Adapt
+{
+  X _x;
+  using Access = AccessTraits<X, Tag>;
+
+public:
+  Adapt(Tag, X const &x)
```

Comment:
Is the only reason you pass `Tag` here for a deduction guide?

Reply:
If by that you mean so that CTAD kicks in, yes.